aptos-move/block-executor/src/executor.rs

```diff
@@ -653,7 +648,15 @@ where
         let finalized_groups = last_input_output.take_finalized_group(txn_idx);
         let materialized_finalized_groups =
             map_id_to_values_in_group_writes(finalized_groups, &latest_view)?;
-        let serialized_groups = serialize_groups::<T>(materialized_finalized_groups, txn_idx)?;
+
+        fail_point!("commit-all-halt-err", |_| Err(code_invariant_error(
```

Comment:
I remember why this fail point was where it was!
the idea was to make sure that even if we execute the block successfully, and halt the scheduler, we can still return with error from parallel execution.
Moving it here makes the error be returned from both transactions (the test has 2), so it is possible to return it even before the whole block is prepared for commit. Maybe we can add a check that txn_idx == scheduler.num_txns() - 1?